src/Sylius/Bundle/CoreBundle/Tests/Functional/StateMachine/OrderWorkflowTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Functional\StateMachine;
+
+use Sylius\Bundle\CoreBundle\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\Customer;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Order\OrderTransitions;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class OrderWorkflowTest extends KernelTestCase
+{
+    private OrderInterface $order;
+
+    public function setup(): void
+    {
+        parent::setUp();
+        $sequenceRepository = $this->createMock(RepositoryInterface::class);
+        $sequenceRepository
+            ->method('findOneBy')
+            ->willReturn(null)
+        ;
+
+        self::getContainer()->set('sylius.repository.order_sequence', $sequenceRepository);
+
+        $channel = $this->createMock(ChannelInterface::class);
+        $customer = $this->createMock(Customer::class);
+        $order = new Order();
+        $order->setChannel($channel);
+        $order->setCustomer($customer);
+
+        $this->order = $order;
+    }
+
+    /** @test */
+    public function it_applies_new_state_for_order_cart_state(): void
+    {
+        $stateMachine = $this->getStateMachine();
+        $order = $this->order;
+
+        $stateMachine->apply($order, OrderTransitions::GRAPH, OrderTransitions::TRANSITION_CREATE);
+
+        $this->assertSame('new', $order->getState());
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider availableTransitionsForNewState
+     */
+    public function it_applies_all_available_transitions_for_order_new_state(
+        string $transition,
+        string $expectedState,
+    ): void {
+        $stateMachine = $this->getStateMachine();
+        $order = $this->order;
+
+        $this->setInitialStateForOrdersNewState();
+        $stateMachine->apply($order, OrderTransitions::GRAPH, $transition);
+
+        $this->assertSame($expectedState, $order->getState());
+    }
+
+    public function availableTransitionsForNewState(): iterable
+    {
+        yield [OrderTransitions::TRANSITION_CANCEL, OrderInterface::STATE_CANCELLED];
+        yield [OrderTransitions::TRANSITION_FULFILL, OrderInterface::STATE_FULFILLED];
+    }
+
+    private function setInitialStateForOrdersNewState(): void
+    {
+        $stateMachine = $this->getStateMachine();
+        $order = $this->order;
+
+        $stateMachine->apply($order, OrderTransitions::GRAPH, OrderTransitions::TRANSITION_CREATE);
+    }
```

Comment:
This is a bit redundant since you can just set the initial state using the data provider above, and have 1 test instead of making a special case out of `cart` => `new`